internal/juju/applications.go

```diff
@@ -1226,6 +1227,24 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 		}
 	}
 
+	if input.UnsetConfig != nil {
+		// these are config entries to be unset
+		if len(input.UnsetConfig) > 0 {
+			c.Debugf("Detected config keys to be unset..")
+			keys := make([]string, 0, len(input.UnsetConfig))
+			for key := range input.UnsetConfig {
+				keys = append(keys, key)
+			}
+			if len(keys) > 0 {
```

Comment:
todo: given that this must be true per line 1232, either remove or add a `else` with an error.